raster_api/runtime/src/alternate_reader.py

```diff
@@ -0,0 +1,55 @@
+"""Custom Alternate Asset Reader."""
+
+import attr
+from rio_tiler.errors import InvalidAssetName, MissingAssets
+from rio_tiler.types import AssetInfo
+
+from titiler.pgstac.reader import PgSTACReader
+
+
+@attr.s
+class PgSTACReaderAlt(PgSTACReader):
+    """Custom STAC Reader for the alternate asset format used widely by NASA.
+
+    Only accept `pystac.Item` as input (while rio_tiler.io.STACReader accepts url or pystac.Item)
+
+    """
+
+    def _get_asset_info(self, asset: str) -> AssetInfo:
+        """Validate asset names and return asset's url.
+        Args:
+            asset (str): STAC asset name.
+        Returns:
+            str: STAC asset href.
+        """
+        if asset not in self.assets:
+            raise InvalidAssetName(f"{asset} is not valid")
+
+        asset_info = self.input.assets[asset]
+        extras = asset_info.extra_fields
+
+        if ("alternate" not in extras) or ("s3" not in extras["alternate"]):
+            raise MissingAssets("No alternate asset found")
+
+        info = AssetInfo(url=extras["alternate"]["s3"]["href"], metadata=extras)
```

Comment:
in rio-tiler we use pystac to make sure the HREF is not relative https://github.com/cogeotiff/rio-tiler/blob/main/rio_tiler/io/stac.py#L298 but this being `custom` I think using direct link is fine 👍